### DISCUSSION TITLE
Wallpaper changer by time of day

### DISCUSSION THREAD

**Discussion by Acheron17** [linuxmint/discussions#499] [Ideas]
It would really be cool to build into Cinnamon a wallpaper changer where the user could specify beginning and ending times of the day to display different wallpapers.  Right now i have to do it manually by clicking on a desktop item in the start menu that calls a script to input gsettings to change the wallpaper.  Could even offer the same feature for themes.

![image](https://github.com/linuxmint/discussions/assets/45599505/4153b24b-a214-479c-96bb-f5ec1f239e1b)

**Comment by rcalixte** [comment]
This has already been implemented as an extension: [Cinnamon Dynamic Wallpaper](https://cinnamon-spices.linuxmint.com/extensions/view/97)

You should be able to install this on your system assuming it is running a supported version of Cinnamon. You should also find the requested settings available.

**Reply by Acheron17** [reply]
I understand.  It would be nice to have it 'built in" as well as dynamic themes.

**Reply by rcalixte** [reply]
I see multiple disadvantages to that without much in the way of advantages but I'm open to hearing a different perspective.

1. As a third-party extension, it can be updated when necessary rather than every six months when Cinnamon is released. This means that translation updates, bug fixes, and general improvements are available right away for users to consume.
2. It is available from the Extensions settings dialog, the same as the built-in ones (of which there are currently not any). The only difference is that it requires an internet connection for the initial installation.
![image](https://github.com/linuxmint/discussions/assets/10281587/007bb7f6-7713-4d08-a0ad-39297f19ccee)
3. Not every user wants or needs this feature so it may not make sense to be built in when it offers little gain but will require development effort and time as opposed to the already completed counterpart.

What would you propose as the benefits or advantages?

**Reply by Acheron17** [reply]
GMOME already has the feature to change the theme based on the user's local sunrise and sunset built into the latest version.  Honestly, if I had known this suggestion would have met so much resistance, I probably wouldn't have bothered with it.  My only issue with third party extensions is stability, maintenance and timely updates with new desktop versions.  This is a nice extension and I will use it while checking its long term stability and how it fares with the jump to Linux Mint 22.  Thanks Rick.

**Reply by rcalixte** [reply]
> Honestly, if I had known this suggestion would have met so much resistance, I probably wouldn't have bothered with it.

This is a genuine conversation. I would like to learn more about your position.



> My only issue with third party extensions is stability, maintenance and timely updates with new desktop versions. This is a nice extension and I will use it while checking its long term stability and how it fares with the jump to Linux Mint 22.

This might be a bit of a perception issue. Stability, maintenance, and updates are not any more guaranteed with one than the other. In the case of the extension here, the developer/maintainer is very active and attentive which is why I usually recommend it.

In the event that an applet or extension falls into an unmaintained state, it is unfortunate but I also don't know how to address something like that long-term. Generally, where there is interest, an interested party appears to take ownership. That is what allows for community-based projects to work. We have to have some trust.

**Reply by rcalixte** [reply]
I'll add a cc for @tobizog here in case there's any interest in developing/maintaining a new extension for the theme-switching equivalent to complement the wallpaper-switching.
